src/docs/arch-schema.dox

```diff
@@ -1,11 +1,117 @@
+ more details.
 
+Column store requires the key format to be defined as the record number \c 'r' type.
+@ref schema has more information on key/value formats.
+
+@section schema_files Data Files
+
+Database schema defines how data files are organized in the database home folder:
+ - A row-oriented table keeps all the data in one file called \c "<table name>.wt", where
+ \c "<table name>" is the name that was passed as a part of \c name parameter to WT_SESSION::create.
+ - A column-oriented table stores the data in multiple files. One for each column group. Filename
+ will be as follows \c "<table name>_<colgroup name>.wt". Where \c "<table name>" is the name that
+ was specified as a part of \c name parameter to WT_SESSION::create. And \c "<colgroup name>" is the
+ column group name defined in \c colgroups entry during the definition of the table format.
+ See this example of how column groups can be configured in WiredTiger @ex_ref{ex_col_store.c}.
+ @ref arch-row-column describes in more detail how row and column stores work.
+ - Each table index is stored in a separate file \c "<table name>_<index name>.wti".
+ Where \c "<table name>" is the table name passed into WT_SESSION::create. And \c "<index name>"
+ is the index name defined during index creation. See @ref schema_indices for more
+ information on how to create a table index.
+ - LSM trees are store on the file system in \c "<table name>-<chunk id>.lsm" files. Where
+ \c "<table name>" is the name that was specified as a part of \c name parameter to
+ WT_SESSION::create. \c "<chunk id>" is the chunk index managed by WiredTiger. More information
+ on LSM trees can be found on this page @ref lsm.
+ - Bloom filters for LSM trees are stored in the files with the same name as the LSM tree chunk,
+ but with different extension: "<table name>-<chunk id>.bf". See @ref lsm_bloom for more details.
+
+@section schema_integrity Schema Integrity
+
+A user can create and manipulate database objects through the API listed on this page
+@ref schema_api. There are several WiredTiger internal objects such as @ref arch-metadata,
+@ref arch-hs, etc. The schema of those objects is locked and cannot be altered from outside
+of WiredTiger.
+
+Schema operations cause an update to the metadata and are performed under the schema lock to avoid
+concurrent operations on the database schema. The following sequence of steps define a generic schema
+operation:
+
+@plantuml_start{schema_generic.png }
+@startuml{schema_generic.png}
+:A schema operation;
+partition with-schema-lock {
+    :perform operation on underlying data-object;
+    :update metadata-file;
+    :checkpoint and sync metadata;
+}
+stop
+@enduml
+@plantuml_end
+
+Apart from the schema API, the schema lock is necessary for many other
+operations in WiredTiger including the following "heavy" database modifications:
+ - The schema lock wraps \c checkpoint \c prepare to avoid any tables being created or dropped
```

Comment:
Is there some other way to mark this other than \c each word?
@ddanderson do you have any suggestions?

Reply:
Backticks work well, thanks!